### PR TITLE
added db_prefix to namespace indexes for each app so they don't smoosh eachother

### PR DESCRIPTION
**Pull request by dangerousbeans**

### CODE REVIEW

**Comment by mmckegg** [comment]
@dangerousbeans In my opinion, this loses the advantage of having shared indexes at all.

What about instead of a namespace, we prefix the index version?

**Comment by mmckegg** [comment]
Oh sorry @dangerousbeans, just caught up with the thread on ssb about this.

Ah, @dominictarr recommend you do it this way. Hmm, I think that your initial idea about version number is much more useful, but that it should be a config option (if someone just uses Patchwork, there is no advantage to keeping old versions around).

File: index.js [hunk 1]
```diff
@@ -101,6 +104,7 @@ var SSB = {
       //temporary!
       _flumeUse                :
         function (name, flumeview) {
+          name = opts.db_prefix + name
```
Comment (christianbundy) [inline index.js]: I might suggest:

```suggestion
          name = opts.dbPrefix + name
```

Comment (dangerousbeans) [reply index.js]: 👍

**Comment by dominictarr** [comment]
@dangerousbeans please try and run this before you make the PR, and you'll see what the problem is ;)

**Comment by dangerousbeans** [comment]
@dominictarr it's running - as far as I can tell it works alright?

**Comment by dominictarr** [comment]
oh you fixed it. @dangerousbeans if you put a / in it will put everything inside a directory, much tidier. `prefix ? path.join(prefix, name) : name` also, please make the option camelCase. There are some snake_case things, but only internal variables and c bindings. please keep the api consistent. or maybe it should be `{flume: {prefix: dbPrefix}}` we havn't needed another flume config, but we should probably keep them together.

**Comment by dangerousbeans** [comment]
@dominictarr should we just namespace via the npm version of flume, or whatever package gets upset? (is that available in scope somewhere?)

**Comment by christianbundy** [comment]
Is it just me, or will this have very strange behavior when `opts.dbName` is undefined?

```javascript
const opts = {}
const name = 'patchbutt'
opts.dbName + name // => "undefinedpatchbutt"

**Comment by christianbundy** [comment]
Hasn't been active for over a year, I'm going to close. Please re-open if you'd like to see this merged!

**Pull request closed without merging**